homeassistant/components/mqtt/client.py

```diff
@@ -416,6 +423,27 @@ async def async_stop_mqtt(_event: Event) -> None:
             hass.bus.async_listen_once(EVENT_HOMEASSISTANT_STOP, async_stop_mqtt)
         )
 
+    def register_naming_issues(self) -> None:
+        """Register issues with MQTT entity naming."""
+        mqtt_data = get_mqtt_data(self.hass)
+        for issue_key, items in mqtt_data.issues.items():
+            config_list = ""
+            for item in items:
+                config_list.join(f"- {item}\n")
```

Comment:
```suggestion
                config_list = config_list.join(f"- {item}\n")
```

Reply:
The argument of join can be a set.

Replace
```
config_list = ""
for item in items:
    config_list.join(f"- {item}\n")
```
by
`config_list = "\n".join(items)`